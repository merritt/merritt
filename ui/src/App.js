import React, { Component } from "react";
import axios from "axios";
import logo from "./logo.svg";
import "./App.css";

class App extends Component {
  state = {
    info: null,
    error: null
  };

  componentDidMount() {
    axios
      .get("/info")
      .then(response => {
        this.setState({
          info: response,
          error: null
        });
      })
      .catch(error => {
        this.setState({
          error: error.message
        });
      });
  }

  render() {
    const { info, error } = this.state;
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>Welcome to Merritt</h2>
        </div>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        {!error && !info && <p className="App-loading">Loading...</p>}
        {error && <p className="App-error">{error}</p>}
        {info &&
          <pre className="App-info">{JSON.stringify(info, null, 2)}</pre>}
      </div>
    );
  }
}

export default App;
