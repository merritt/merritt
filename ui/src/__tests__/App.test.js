import React from "react";
import { mount, shallow } from "enzyme";
import App from "../App";

describe("App", () => {
  it("shallow renders without crashing", () => {
    shallow(<App />);
  });

  it("matches snapshot", () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    mount(<App />);
  });

  it("renders welcome message", () => {
    const wrapper = shallow(<App />);
    const welcome = <h2>Welcome to Merritt</h2>;
    expect(wrapper).toContainReact(welcome);
  });
});
