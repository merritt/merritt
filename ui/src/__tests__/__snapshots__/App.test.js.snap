exports[`App matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="App">
    <div
        className="App-header">
        <img
            alt="logo"
            className="App-logo"
            src="logo.svg" />
        <h2>
            Welcome to Merritt
        </h2>
    </div>
    <p
        className="App-intro">
        To get started, edit 
        <code>
            src/App.js
        </code>
         and save to reload.
    </p>
    <p
        className="App-loading">
        Loading...
    </p>
</div>,
  "nodes": Array [
    <div
      className="App">
      <div
            className="App-header">
            <img
                  alt="logo"
                  className="App-logo"
                  src="logo.svg" />
            <h2>
                  Welcome to Merritt
            </h2>
      </div>
      <p
            className="App-intro">
            To get started, edit 
            <code>
                  src/App.js
            </code>
             and save to reload.
      </p>
      <p
            className="App-loading">
            Loading...
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "error": null,
          "info": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="App">
          <div
                    className="App-header">
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg" />
                    <h2>
                              Welcome to Merritt
                    </h2>
          </div>
          <p
                    className="App-intro">
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
          <p
                    className="App-loading">
                    Loading...
          </p>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="App">
          <div
                    className="App-header">
                    <img
                              alt="logo"
                              className="App-logo"
                              src="logo.svg" />
                    <h2>
                              Welcome to Merritt
                    </h2>
          </div>
          <p
                    className="App-intro">
                    To get started, edit 
                    <code>
                              src/App.js
                    </code>
                     and save to reload.
          </p>
          <p
                    className="App-loading">
                    Loading...
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
